jina/serve/runtimes/request_handlers/worker_request_handler.py

```diff
@@ -346,26 +350,32 @@ def close(self):
             self._is_closed = True
 
     @staticmethod
-    def get_docs_matrix_from_request(
+    def _get_docs_matrix_from_request(
         requests: List['DataRequest'],
-        field: str,
-    ) -> List['DocumentArray']:
+        return_map: bool = True,
+    ) -> Tuple[Optional[List['DocumentArray']], Optional[Dict[str, 'DocumentArray']]]:
         """
         Returns a docs matrix from a list of DataRequest objects.
+
         :param requests: List of DataRequest objects
-        :param field: field to be retrieved
-        :return: docs matrix: list of DocumentArray objects
+        :param return_map: bool indicating if a map should be returned
+        :return: docs matrix and doc: list of DocumentArray objects
         """
+        docs_map = None
         if len(requests) > 1:
-            result = [getattr(request, field) for request in requests]
+            docs_matrix = [request.docs for request in requests]
+            if return_map:
+                docs_map = {req.last_executor: req.docs for req in requests}
```

Comment:
I would construct both docs_matrix and docs_map in the same for loop

Reply:
good point